test/e2e/proxy_test.go

```diff
@@ -181,82 +263,113 @@ func testProxyLoadBalancerService(t *testing.T, isIPv6 bool) {
+			for idx, node := range nodes {
+				createAgnhostPod(t, data, hostAgnhostsCluster[idx], node, true, agnhostLBClusterServiceLabel)
+			}
+			defer func() {
+				for idx := range nodes {
+					require.NoError(t, data.deletePod(testNamespace, hostAgnhostsCluster[idx]))
+				}
+			}()
+			loadBalancerClusterTestCases(t, data, clusterUrl, nodes, clients, clientOSes)
+		})
+		t.Run("ExternalTrafficPolicy:Local", func(t *testing.T) {
+			t.Parallel()
+			port := "8081"
+			localUrl := net.JoinHostPort(localIngressIP[0], port)
+			hostAgnhostsCluster := []string{"endpoint-lb-host-local-0", "endpoint-lb-host-local-1"}
+			for idx, node := range nodes {
+				createAgnhostPod(t, data, hostAgnhostsCluster[idx], node, true, agnhostLBLocalServiceLabel)
+			}
+			defer func() {
+				for idx := range nodes {
+					require.NoError(t, data.deletePod(testNamespace, hostAgnhostsCluster[idx]))
+				}
+			}()
+			loadBalancerLocalTestCases(t, data, localUrl, nodes, nodeIPs, clients, clientIPs, clientOSes, true)
+		})
 	})
 }
 
 func testLoadBalancerClusterFromNode(t *testing.T, data *TestData, nodes []string, url string) {
 	skipIfKubeProxyEnabled(t, data)
 	for _, node := range nodes {
-		require.NoError(t, probeFromNode(node, url), "Service LoadBalancer whose externalTrafficPolicy is Cluster should be able to be connected from Node")
+		require.NoError(t, probeFromNode(node, url),
+			"Service LoadBalancer should be able to be connected from Node '%s' with URL '%s'", node, url)
 	}
 }
 
-func testLoadBalancerClusterFromPod(t *testing.T, data *TestData, pods []string, url string) {
-	for _, pod := range pods {
-		require.NoError(t, probeFromPod(data, pod, url), "Service LoadBalancer whose externalTrafficPolicy is Cluster should be able to be connected from Pod")
+func testLoadBalancerClusterFromPod(t *testing.T, data *TestData, pods, podOSes []string, url string) {
+	for idx, pod := range pods {
+		require.NoError(t, probeFromPod(data, pod, podOSes[idx], url),
+			"Service LoadBalancer should be able to be connected from Pod '%s' with url '%s'", pod, url)
 	}
 }
 
-func testLoadBalancerLocalFromNode(t *testing.T, data *TestData, nodes []string, url string, expectedHostnames []string) {
+func testLoadBalancerLocalFromNode(t *testing.T, data *TestData, nodes []string, url string, expectedHostNetwork string) {
 	skipIfKubeProxyEnabled(t, data)
-	for idx, node := range nodes {
-		hostname, err := probeHostnameFromNode(node, url)
-		require.NoError(t, err, "Service LoadBalancer whose externalTrafficPolicy is Local should be able to be connected from Node")
-		require.Equal(t, hostname, expectedHostnames[idx])
+	for _, node := range nodes {
+		msg := fmt.Sprintf("Service LoadBalancer should be able to be connected from Node '%s' with url '%s/%s'", node, url, hostNetworkNodeEP)
+		hostNetwork, err := probeEndpointFromNode(node, url, hostNetworkNodeEP)
+		require.NoError(t, err, msg)
+		assert.Equal(t, expectedHostNetwork, hostNetwork, msg)
```

Comment:
I think the original test checks the hostname to ensure the backend Pod is local to the accessed Node. I don't get how checking hostNetwork achieves the same purpose. Even if the Local externalTrafficPolicy is implemented wrongly, the test cannot detect it.
@hongliangl may have insight on this.

Reply:
Do you have suggestion to this comment? A good way to get container id without docker? https://github.com/antrea-io/antrea/pull/2950#discussion_r742226997

Reply:
@lzhecheng currently you are using an environment variable to export whether or not the Pod runs in the host network. Could you do the same thing for the Node name using the downward API:
```yaml
      env:
        - name: nodeName
          valueFrom:
            fieldRef:
              fieldPath: spec.nodeName
```
Then you can query the following URL: `/shell?cmd=echo+$nodeName`
Would that let you implement a test that checks the correct implementation of `externalTrafficPolicy`?